op-acceptor/runner/logging_test.go

```diff
@@ -68,3 +70,133 @@ gates:
+        tests:
+          - name: TestLogLevelEnvironment
+            package: "./main"
+`)
+
+	r := setupTestRunner(t, testContent, configContent)
+	r.testLogLevel = "debug"
+	err := os.WriteFile(filepath.Join(r.workDir, "main_test.go"), testContent, 0644)
+	require.NoError(t, err)
+
+	result, err := r.RunTest(ctx, types.ValidatorMetadata{
+		ID:       "test1",
+		Gate:     "logging-gate",
+		FuncName: "TestLogLevelEnvironment",
+		Package:  ".",
+	})
+
+	require.NoError(t, err)
+	assert.Equal(t, types.TestStatusPass, result.Status)
+	assert.Equal(t, "test1", result.Metadata.ID)
+	assert.Equal(t, "logging-gate", result.Metadata.Gate)
+	assert.Equal(t, ".", result.Metadata.Package)
+	assert.False(t, result.Metadata.RunAll)
+	assert.Contains(t, result.Stdout, "TEST_LOG_LEVEL set to debug")
+}
+
+// TestOutputTestLogs verifies that test logs are properly captured and output when enabled
+func TestOutputTestLogs(t *testing.T) {
+	// Create a test file that outputs logs at different levels
+	testContent := []byte(`
+package feature_test
+
+import (
+	"testing"
+)
+
+func TestWithLogs(t *testing.T) {
+	t.Log("This is a test output")
+}
+`)
+
+	configContent := []byte(`
+gates:
+  - id: logging-gate
+    description: "Gate with a test that outputs logs"
+    suites:
+      logging-suite:
+        description: "Suite with a test that outputs logs"
+        tests:
+          - name: TestWithLogs
+            package: "./feature"
+`)
+
+	// Setup the test runner with OutputTestLogs enabled
+	r := setupTestRunner(t, testContent, configContent)
+	r.outputTestLogs = true
+
+	// Run the test
+	result, err := r.RunAllTests(context.Background())
+	require.NoError(t, err)
+	assert.Equal(t, types.TestStatusPass, result.Status)
+
+	// Verify the structure
+	require.Contains(t, result.Gates, "logging-gate")
+	gate := result.Gates["logging-gate"]
+	require.Contains(t, gate.Suites, "logging-suite")
+	suite := gate.Suites["logging-suite"]
+
+	// Get the test result
+	var testResult *types.TestResult
+	for _, test := range suite.Tests {
+		testResult = test
+		break
+	}
+	require.NotNil(t, testResult)
+
+	// Verify the test passed and has logs captured
+	assert.Equal(t, types.TestStatusPass, testResult.Status)
+	assert.NotEmpty(t, testResult.Stdout)
+	assert.Contains(t, testResult.Stdout, "This is a test output")
+
+	// Now run with OutputTestLogs disabled
+	r.outputTestLogs = false
+	result, err = r.RunAllTests(context.Background())
+	require.NoError(t, err)
+	assert.Equal(t, types.TestStatusPass, result.Status)
+
+	// Get the test result again
+	var testResult2 *types.TestResult
+	for _, test := range result.Gates["logging-gate"].Suites["logging-suite"].Tests {
+		testResult2 = test
+		break
+	}
+	require.NotNil(t, testResult)
+
+	// Verify that logs are not captured when OutputTestLogs is disabled
+	assert.Equal(t, types.TestStatusPass, testResult2.Status)
+	assert.Contains(t, testResult2.Stdout, "This is a test output")
```

Comment:
Shouldn't this be asserting that it does NOT contain this output?

```suggestion
	// Verify that logs are not captured when OutputTestLogs is disabled
	assert.Equal(t, types.TestStatusPass, testResult2.Status)
	assert.NotContains(t, testResult2.Stdout, "This is a test output")
```

Reply:
Actually the input is shown either way because we're not saving it to a file, that's what the test itself is testing. I think the other pre-existing tests cover the file logging part so I didn't add for that.

Reply:
I'm afraid I don't understand. How is this block of code verifying that logs are NOT captured?

```
	// Verify that logs are not captured when OutputTestLogs is disabled
	assert.Equal(t, types.TestStatusPass, testResult2.Status)
	assert.Contains(t, testResult2.Stdout, "This is a test output")
```